tests/tracer/test_propagation.py

```diff
@@ -368,6 +373,328 @@ def test_get_wsgi_header(tracer):
+    assert traceparent_values == expected_sampling_priority
+
+
+@pytest.mark.parametrize(
+    "name,headers,expected_tuple,expected_logging,expected_exception",
+    [
+        (
+            "traceflag_01",
+            "00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01",
+            # tp, trace_id, span_id, sampling_priority
+            (11803532876627986230, 67667974448284343, 1),
+            None,
+            None,
+        ),
+        (
+            "traceflag_00",
+            "00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-00",
+            # tp, trace_id, span_id, sampling_priority
+            (11803532876627986230, 67667974448284343, 0),
+            None,
+            None,
+        ),
+        (
+            "unsupported_version",
+            "01-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01",
+            # tp, trace_id, span_id, sampling_priority
+            (11803532876627986230, 67667974448284343, 1),
+            ["unsupported traceparent version:'01', still attempting to parse"],
+            None,
+        ),
+        (
+            "short_version",
+            "0-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01",
+            # tp, trace_id, span_id, sampling_priority
+            (11803532876627986230, 67667974448284343, 1),
+            ["unsupported traceparent version:'0', still attempting to parse"],
+            None,
+        ),
+        (
+            "short_trace_id",
+            "00-f92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01",
+            # tp, trace_id, span_id, sampling_priority
+            None,
+            [
+                "received invalid w3c traceparent: 00-f92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01.",
+                "W3C traceparent hex length incorrect: 00-f92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01",
+            ],
+            ValueError,
+        ),
+        (  # we still parse the trace flag and analyze the it as a bit field
+            "unknown_trace_flag",
+            "00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-02",
+            # tp, trace_id, span_id, sampling_priority
+            (11803532876627986230, 67667974448284343, 0),
+            None,
+            None,
+        ),
+    ],
+)
+def test_extract_traceparent(caplog, name, headers, expected_tuple, expected_logging, expected_exception):
+    with caplog.at_level(logging.DEBUG):
+        if expected_exception:
+            with pytest.raises(expected_exception):
+                traceparent_values = _TraceContext._get_traceparent_values(headers)
+        else:
+            traceparent_values = _TraceContext._get_traceparent_values(headers)
+            assert traceparent_values == expected_tuple
+            if caplog.text:
+                for expected_log in expected_logging:
+                    assert expected_log in caplog.text
+
+
+@pytest.mark.parametrize(
+    "name,ts_string,expected_tuple,expected_logging,expected_exception",
```

Comment:
Same as above though the names here provide a reader useful information. In this case, we could replace them with the `id` kwargs.

Reply:
Why should we replace them with the id kwargs? In my mind it seems like it'd be easier to tell what a test is testing if the name is included.

Reply:
nvm, thanks for explaining in person